python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1489,14 +1574,38 @@ def _make_outputs(self, wcsf, visitSummaryTables, exposureInfo, mapTemplate=None
                 Initial WCS for this extension.
             ``extensionType`` : `np.ndarray` [`str`]
                 "SCIENCE" or "REFERENCE".
+        mapTemplate : `dict` [`str`, `str`]
+            Dictionary containing the model description.
+        inputCameraModel : `dict` [`str`, `np.ndarray`], optional
+            Parameters to use for the device part of the model. This must be
+            provided if an input camera model was used.
+
         Returns
         -------
         catalogs : `dict` [`str`, `lsst.afw.table.ExposureCatalog`]
             Dictionary of `lsst.afw.table.ExposureCatalog` objects with the WCS
             set to the WCS fit in wcsf, keyed by the visit name.
+        cameraParams : `dict` [`str`, `np.ndarray`], optional
+            Parameters for the device part of the model in the format needed
+            when used as input for future runs.
         """
         # Get the parameters of the fit models
         mapParams = wcsf.mapCollection.getParamDict()
+        cameraParams = {}
+        if self.config.saveCameraModel:
+            for element in mapTemplate["INSTRUMENT/DEVICE"]["Elements"]:
+                for detector in exposureInfo.detectors:
+                    detectorTemplate = element.replace("DEVICE", str(detector))
+                    detectorTemplate = detectorTemplate.replace("BAND", ".+")
+                    for k, params in mapParams.items():
+                        if re.fullmatch(detectorTemplate, k):
+                            cameraParams[k] = params
+        if self.config.useInputCameraModel:
+            if inputCameraModel is None:
+                raise RuntimeError(
+                    "inputCameraModel must be provided to _make_outputs in order to build" " output WCS."
+                )
+            mapParams.update(inputCameraModel)
```

Comment:
After some thought, I decided that the task should raise an error if both `useInputCameraModel` and `saveCameraModel` are true. Hypothetically you could carry through the input camera model and save it, but it would not work as currently coded because the camera parameters are not included in the `mapParams` if they are not fit. Also, I think it would confuse the true provenance of the camera model.

Reply:
Ah, I was thinking maybe you could input an "initial guess" at the camera model, refine it, and then output a new model.  But prohibiting using both input and save for reasons of provenance probably is safer.